### PR TITLE
MWPW-139379 Improve Tab Scrolling for Mobile

### PR DESCRIPTION
**Pull request by Brandon32**
* Improve mobile experience for tab scrolling by adding scroll indicator buttons. - reverted by issue with geo routing 
  * #1624
  * #1733
* Fixes geo routing tabs 

Resolves: [MWPW-139379](https://jira.corp.adobe.com/browse/MWPW-139379), [MWPW-141245](https://jira.corp.adobe.com/browse/MWPW-141245)

Test URLs:

- Before: https://main--milo--adobecom.hlx.page/drafts/bmarshal/tabs-scrolling?martech=off
- After: https://bmarshal-tabs-scrolling-2--milo--adobecom.hlx.page/drafts/bmarshal/tabs-scrolling?martech=off
- Before: https://main--milo--adobecom.hlx.page/drafts/bmarshal/tabs-scrolling-dark?martech=off
- After: https://bmarshal-tabs-scrolling-2--milo--adobecom.hlx.page/drafts/bmarshal/tabs-scrolling-dark?martech=off
- Before: https://main--milo--adobecom.hlx.page/drafts/bmarshal/tabs-scrolling-quiet-transparent?martech=off
- After: https://bmarshal-tabs-scrolling-2--milo--adobecom.hlx.page/drafts/bmarshal/tabs-scrolling-quiet-transparent?martech=off
- Before: https://main--milo--adobecom.hlx.page/drafts/bmarshal/tabs-scrolling-pill?martech=off
- After: https://bmarshal-tabs-scrolling-2--milo--adobecom.hlx.page/drafts/bmarshal/tabs-scrolling-pill?martech=off

BACOM URLs:

- Before: https://main--bacom--adobecom.hlx.page/?martech=off
- After: https://main--bacom--adobecom.hlx.page/?martech=off&milolibs=bmarshal-tabs-scrolling-2
- Before: https://main--bacom--adobecom.hlx.page/?georouting=on&akamaiLocale=BE
- After: https://main--bacom--adobecom.hlx.page/?georouting=on&akamaiLocale=BE&milolibs=bmarshal-tabs-scrolling-2

### CODE REVIEW

File: libs/blocks/tabs/tabs.css [hunk 1]
```diff
@@ -184,6 +185,7 @@
   background: unset;
 }
 
+
```
Comment (meganthecoder) [inline libs/blocks/tabs/tabs.css:188]: Remove extra line

File: test/blocks/tabs/tabs.test.js [hunk 2]
```diff
@@ -35,4 +42,58 @@ describe('tabs', () => {
...
+    expect(leftPaddle.getAttribute('disabled')).to.equal('');
+    expect(rightPaddle.getAttribute('disabled')).to.equal('');
+  });
+
+  it('right paddle is visible when tabList is scrollable', async () => {
+    setViewport({ width: MOBILE_WIDTH, height: HEIGHT });
+    window.dispatchEvent(new Event('resize'));
+    const rightPaddle = allTabs[0].querySelector('.paddle-right');
+
+    console.log(rightPaddle.getAttribute('disabled'));
```
Comment (meganthecoder) [inline test/blocks/tabs/tabs.test.js:62]: Remove console.log

File: test/blocks/tabs/tabs.test.js [hunk 3]
```diff
@@ -35,4 +42,58 @@ describe('tabs', () => {
     await sendKeys({ press: 'Space' });
     expect(unSelectedBtn2.ariaSelected).to.equal('false');
   });
+
+  it('paddles are disabled when tabList is not scrollable', async () => {
```
Comment (meganthecoder) [inline test/blocks/tabs/tabs.test.js:46]: Tests should start with a verb and make sense as a sentence
```suggestion
  it('disables paddles when tabList is not scrollable', async () => {
```

**Review by meganthecoder — APPROVED** [review]
Looks good! Just a couple of small things

File: test/blocks/tabs/tabs.test.js (hunk 3 above)
Comment (Brandon32) [reply test/blocks/tabs/tabs.test.js]: Thanks

File: test/blocks/tabs/tabs.test.js (hunk 2 above)
Comment (Brandon32) [reply test/blocks/tabs/tabs.test.js]: Removed

File: libs/blocks/tabs/tabs.css (hunk 1 above)
Comment (Brandon32) [reply libs/blocks/tabs/tabs.css]: Removed

File: test/blocks/tabs/tabs.test.js [hunk 4]
```diff
@@ -35,4 +42,57 @@ describe('tabs', () => {
...
+    expect(leftPaddle.getAttribute('disabled')).to.equal('');
+    expect(rightPaddle.getAttribute('disabled')).to.equal('');
+  });
+
+  it('right paddle is visible when tabList is scrollable', async () => {
+    setViewport({ width: MOBILE_WIDTH, height: HEIGHT });
+    window.dispatchEvent(new Event('resize'));
+    const rightPaddle = allTabs[0].querySelector('.paddle-right');
+
+    expect(rightPaddle.getAttribute('disabled')).to.equal('');
```
Comment (JasonHowellSlavin) [inline test/blocks/tabs/tabs.test.js:62]: is this the correct matcher? Based on the descriptive text, I'd expect a match against visibility, like the display property or opacity. Also, seems like the right paddle should not be disabled when scrollable.

File: libs/blocks/tabs/tabs.js [hunk 5]
```diff
@@ -4,36 +4,39 @@
...
-const isElementInContainerView = (targetEl) => {
-  const rect = targetEl.getBoundingClientRect();
-  const docEl = document.documentElement;
-  return (
-    rect.top >= 0
-    && rect.left >= 0
-    && rect.bottom <= (window.innerHeight || /* c8 ignore next */ docEl.clientHeight)
-    && rect.right <= (window.innerWidth || /* c8 ignore next */ docEl.clientWidth)
-  );
+const RIGHT_PADDLE = '<svg viewBox="0 0 8 14" fill="none" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" clip-rule="evenodd" d="M1.50001 13.25C1.22022 13.25 0.939945 13.1431 0.726565 12.9292C0.299315 12.5019 0.299315 11.8096 0.726565 11.3823L5.10938 7L0.726565 2.61768C0.299315 2.19043 0.299315 1.49805 0.726565 1.0708C1.15333 0.643068 1.84669 0.643068 2.27345 1.0708L7.4297 6.22656C7.63478 6.43164 7.75001 6.70996 7.75001 7C7.75001 7.29004 7.63478 7.56836 7.4297 7.77344L2.27345 12.9292C2.06007 13.1431 1.7798 13.2495 1.50001 13.25Z" fill="currentColor"/></svg>';
```
Comment (JasonHowellSlavin) [inline libs/blocks/tabs/tabs.js:7]: nit: Maybe just `PADDLE`. Though I get why using `RIGHT_PADDLE` since its initial state is right facing.

**Review by JasonHowellSlavin — APPROVED** [review]
Small nits, looks good.

File: test/blocks/tabs/tabs.test.js (hunk 4 above)
Comment (Brandon32) [reply test/blocks/tabs/tabs.test.js:62]: Yes this is the correct matcher

File: libs/blocks/tabs/tabs.js (hunk 5 above)
Comment (Brandon32) [reply libs/blocks/tabs/tabs.js:7]: Being more descriptive is better

File: libs/blocks/tabs/tabs.css [hunk 6]
```diff
@@ -25,6 +26,7 @@
   --tabs-pill-bg-color-hover: #666;
   --tabs-pill-bg-color-active: #444;
   --tabs-pill-text-color: #fff;
+  --tabs-paddle-bs-color: #ffffff1a;
```
[the remark this answers is not in the document]
Comment (Brandon32) [reply libs/blocks/tabs/tabs.css:29]: Hex is more convenient and consistent, if we don't want alpha in hex values then it should be disallowed.
https://stylelint.io/user-guide/rules/color-hex-alpha#never

File: libs/blocks/tabs/tabs.js [hunk 7]
```diff
@@ -84,6 +87,77 @@ function initTabs(elm, config, rootElem) {
   if (config) configTabs(config, rootElem);
 }
 
+// Check if current tab is not visible and next tab is visible
+function previousTab(item, i, arr) {
+  const next = arr[i + 1];
+  return (!isTabInTabListView(item) && next && isTabInTabListView(next));
```
Comment (chrischrischris) [inline libs/blocks/tabs/tabs.js:93]: nit: `next` should be first in check so if false we don't run `isTabInTabListView`

File: libs/blocks/tabs/tabs.js [hunk 8]
```diff
@@ -84,6 +87,77 @@ function initTabs(elm, config, rootElem) {
...
+}
+
+// Check if previous tab is visible and current tab is not visible
```
Comment (chrischrischris) [inline libs/blocks/tabs/tabs.js:96]: Nit: this comment doesn't seem to apply to the `nextTab` function - though I know it's probably for the block of code below it.  I don't think it's needed.

**Review by chrischrischris — APPROVED** [review]

**Review by ryanmparrish — APPROVED** [review]
Looks good. Nice work!
Have a look at those nit's otherwise gtg. 👍

File: libs/blocks/tabs/tabs.js (hunk 7 above)
Comment (Brandon32) [reply libs/blocks/tabs/tabs.js]: Updated.

File: libs/blocks/tabs/tabs.js (hunk 8 above)
Comment (Brandon32) [reply libs/blocks/tabs/tabs.js]: I re-wrote and simplified this comment, there was some confusion in the last PR and want to keep some explanation.

**Pull request merged**